### PR TITLE
Update references for conformance

### PR DESCRIPTION
**Pull request by smaye81**
This renames all references and usages to the old crosstest repo so that they begin using the new `connectrpc/conformance` repository in the `connectrpc` organization.

Note this is not intended to merge (or build successfully) until the crosstest repo is officially moved.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,9 +155,9 @@ for details.
 
 - [connect-kotlin][connect-kotlin]: Idiomatic gRPC & Connect RPCs for Kotlin
 - [connect-go][connect-go]: Go service stubs for servers
-- [connect-web][connect-web]: TypeScript clients for web browsers
+- [connect-es][connect-es]: Type-safe APIs with Protobuf and TypeScript
 - [Buf Studio][buf-studio]: Web UI for ad-hoc RPCs
-- [connect-crosstest][connect-crosstest]: Connect, gRPC, and gRPC-Web
+- [conformance][connect-conformance]: Connect, gRPC, and gRPC-Web
```
Comment (rebello95) [inline README.md:160]:
```suggestion
- [connect-conformance][connect-conformance]: Connect, gRPC, and gRPC-Web
```

**Review by rebello95 — APPROVED** [review]
Looks good, assuming you've grepped the whole codebase for `crosstest` as well

File: README.md (hunk 1 above)
Comment (smaye81) [reply README.md:160]: It was actually decided to use just `conformance` for this link (see [here](https://github.com/bufbuild/connect-crosstest/pull/539#discussion_r1313260511) and [here](https://github.com/connectrpc/connect-go/pull/579#discussion_r1313672056)). But I can make the link reference also be `conformance` for consistency if you like?

Comment (rebello95) [reply README.md:160]: I don't have a super strong opinion, good to keep it consistent across repos

**Comment by smaye81** [comment]
Closing this because I'm going to split this up into PRs.

**Pull request closed without merging**